testsuite/geb-testsuite/src/test/java/org/sonatype/nexus/geb/GebTestSupport.groovy

```diff
@@ -0,0 +1,562 @@
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.geb
+
+import java.nio.file.Paths
+import java.util.concurrent.Callable
+import java.util.concurrent.TimeUnit
+
+import javax.inject.Inject
+import javax.inject.Named
+import javax.net.ssl.HostnameVerifier
+import javax.net.ssl.HttpsURLConnection
+
+import org.sonatype.nexus.common.app.ApplicationDirectories
+import org.sonatype.nexus.common.event.EventSubscriberHost
+import org.sonatype.nexus.geb.traits.ExtJS
+import org.sonatype.nexus.scheduling.TaskScheduler
+import org.sonatype.sisu.goodies.common.Loggers
+import org.sonatype.sisu.litmus.testsupport.junit.TestDataRule
+import org.sonatype.sisu.litmus.testsupport.junit.TestIndexRule
+import org.sonatype.sisu.litmus.testsupport.port.PortRegistry
+
+import com.google.common.base.Joiner
+import com.google.common.base.Throwables
+import com.google.common.util.concurrent.Uninterruptibles
+import geb.ConfigurationLoader
+import geb.junit4.GebReportingTest
+import org.junit.After
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.ExpectedException
+import org.junit.runner.RunWith
+import org.ops4j.net.URLUtils
+import org.ops4j.pax.exam.Configuration
+import org.ops4j.pax.exam.CoreOptions
+import org.ops4j.pax.exam.MavenUtils
+import org.ops4j.pax.exam.Option
+import org.ops4j.pax.exam.OptionUtils
+import org.ops4j.pax.exam.junit.PaxExam
+import org.ops4j.pax.exam.junit.PaxExamServer
+import org.ops4j.pax.exam.karaf.options.KarafDistributionConfigurationFileExtendOption
+import org.ops4j.pax.exam.options.MavenUrlReference
+import org.ops4j.pax.exam.options.ProvisionOption
+import org.ops4j.pax.exam.options.WrappedUrlProvisionOption.OverwriteMode
+import org.slf4j.Logger
+
+import static com.google.common.base.Preconditions.checkNotNull
+import static org.ops4j.pax.exam.CoreOptions.composite
+import static org.ops4j.pax.exam.CoreOptions.maven
+import static org.ops4j.pax.exam.CoreOptions.mavenBundle
+import static org.ops4j.pax.exam.CoreOptions.propagateSystemProperty
+import static org.ops4j.pax.exam.CoreOptions.systemProperty
+import static org.ops4j.pax.exam.CoreOptions.systemTimeout
+import static org.ops4j.pax.exam.CoreOptions.vmOptions
+import static org.ops4j.pax.exam.CoreOptions.when
+import static org.ops4j.pax.exam.CoreOptions.wrappedBundle
+import static org.ops4j.pax.exam.karaf.options.KarafDistributionOption.*
+
+@RunWith(PaxExam.class)
+//class TestSpec extends GebReportingSpec // TODO - KR would need a pax-exam compatible running for Spock Sputnik here
+class GebTestSupport
```

Comment:
primarily C&P from [NexusPaxExamSupport](https://github.com/sonatype/nexus-oss/blob/gebTest/testsupport/nexus-pax-exam/src/main/java/org/sonatype/nexus/pax/exam/NexusPaxExamSupport.java) with room for Groovyfication